### DISCUSSION TITLE
Apply changes with `--platform=local`

### DISCUSSION THREAD

**Discussion by rahulsom** [renovatebot/renovate#27458] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.202.3

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### What is your question?

I would like to have renovate apply changes locally when using `--platform=local`. Is something like that possible?

E.g.

I would like to specify a config like this

```json
{
  "packageRules": [
    {"matchCategories": ["*"], "enabled": false}, // disable all managers
    {"matchCategories": ["js"], "enabled": true}, // enable only js
    {"groupName": "js-packages", "matchPackagePrefixes":[ "" ]} // collapse all changes into one change
  ]
}
```

Then I would like to invoke renovate like this

```shell
renovate --platform=local --changeset=js-packages
```

The hope is to update all js packages locally.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
This is not supported or planned functionality for platform=local. What's your use case for this?

**Reply by rahulsom** [reply]
I'm trying to make this change a part of other changes that I'm hoping to wrap into one update that's orchestrated outside of renovate.

**Comment by lcarva** [comment]
Having the ability to run renovate locally in an ad-hoc manner so it makes changes to a local repository (without requiring a pull-request workflow) would be very useful and lower the entry-barrier for renovate development.

Would the renovate maintainers be open to this functionality?

**Reply by lcarva** [reply]
Some use cases:

1. More easily hack on the renovate code base. Be able to verify changes to a manger, for example.
2. More easily provide ad-hoc updates. Sometimes it's useful to run renovate right now instead of waiting for the renovate service.

Those are possible today but require going through a pull-request workflow which requires going through the process of getting a GitHub token. This is completely about making those items more convenient.

**Reply by rarkins** [reply]
Open to the idea but we need to make sure the implementation is clean/maintainable
